player/Music_Player.cpp

```diff
@@ -5,6 +5,8 @@
 #include <string.h>
 #include <ctype.h>
 #include "SDL_rwops.h"
+#include "Archive_Reader.h"
+#include "Gme_File.h"
```

Comment:
Is it a good idea to use library's private headers here?

As I understand it, you use it for `gme_type_t_` details:
Are there no alternatives?

Reply:
maybe a function could be added to gme.cpp to perform essentially the same check. i'll look into it

Reply:
Maybe something like this? [patch.txt](https://github.com/libgme/game-music-emu/files/13465943/patch.txt)

```diff
diff --git a/gme/gme.cpp b/gme/gme.cpp
index bb8b52a..6ffccaf 100644
--- a/gme/gme.cpp
+++ b/gme/gme.cpp
@@ -429,3 +429,16 @@ const char* gme_type_system( gme_type_t type )
 	assert( type );
 	return type->system;
 }
+
+int gme_type_trackcnt( gme_type_t type )
+{
+	assert( type );
+	return type->track_count;
+}
+
+int gme_type_same_ctor( gme_type_t a, gme_type_t b )
+{
+	assert( a );
+	assert( b );
+	return a->new_emu == b->new_emu;
+}
diff --git a/gme/gme.exports b/gme/gme.exports
index ea16df7..c9e9ed4 100644
--- a/gme/gme.exports
+++ b/gme/gme.exports
@@ -61,3 +61,5 @@ gme_vgz_type
 gme_disable_echo
 gme_set_fade_msecs
 gme_load_tracks
+gme_type_trackcnt
+gme_type_same_ctor
diff --git a/gme/gme.h b/gme/gme.h
index 89fa4d6..4e3c273 100644
--- a/gme/gme.h
+++ b/gme/gme.h
@@ -232,6 +232,14 @@ BLARGG_EXPORT gme_type_t const* gme_type_list();
 /* Name of game system for this music file type */
 BLARGG_EXPORT const char* gme_type_system( gme_type_t );
 
+/* Number of tracks for format
+ * @since 0.6.4 */
+BLARGG_EXPORT int gme_type_trackcnt( gme_type_t );
+
+/* Whether the formats use the same emulator creator
+ * @since 0.6.4 */
+BLARGG_EXPORT int gme_type_same_ctor( gme_type_t, gme_type_t );
+
 /* True if this music file type supports multiple tracks */
 BLARGG_EXPORT int gme_type_multitrack( gme_type_t );
 
diff --git a/player/Music_Player.cpp b/player/Music_Player.cpp
index 57521ff..4069430 100644
--- a/player/Music_Player.cpp
+++ b/player/Music_Player.cpp
@@ -6,7 +6,6 @@
 #include <ctype.h>
 #include "SDL_rwops.h"
 #include "Archive_Reader.h"
-#include "Gme_File.h"
 
 /* Copyright (C) 2005-2010 by Shay Green. Permission is hereby granted, free of
 charge, to any person obtaining a copy of this software module and associated
@@ -168,12 +167,12 @@ gme_err_t Music_Player::load_file(const char* path , bool by_mem)
 			{ // copy data and file sizes
 				gme_type_t t;
 				RETURN_ERR( in.read( bp ) );
-				if ( (t = gme_identify_extension( in.entry_name() ))
-				&& t->track_count == 1 )
+				if ((t = gme_identify_extension(in.entry_name())) != nullptr &&
+				    gme_type_trackcnt(t) == 1 )
 				{
 					if ( !emu_type )
 						emu_type = t;
-					if ( t->new_emu == emu_type->new_emu )
+					if ( gme_type_same_ctor(t, emu_type) )
 						bp += (sizes[n++] = in.entry_size());
 				}
 			}
```

Reply:
For now, we can leave out the constructor comparison. That won't be useful until we can properly read .vgz files inside of an archive, if that ever ends up happening.

Reply:
OK